src/go/rpk/pkg/cli/cluster/config/set.go

```diff
@@ -60,92 +77,149 @@ You may also use <key>=<value> notation for setting configuration properties:
-			out.MaybeDie(err, "unable to check if the cluster needs to be restarted: %v\nCheck the status with 'rpk cluster config status'.", err)
-			for _, value := range status {
-				if value.Restart {
-					fmt.Print("\nCluster needs to be restarted. See more details with 'rpk cluster config status'.\n")
+		meta, ok := schema[key]
+
+		if !ok {
+			// loop over schema, try to find key in the Aliases,
+			for _, v := range schema {
+				if slices.Contains(v.Aliases, key) {
+					meta, ok = v, true
 					break
 				}
 			}
+			if !ok {
+				return nil, nil, fmt.Errorf("Unknown property %q", key)
+			}
+		}
+
+		// - For scalars, pass string values through to the REST
+		// API -- it will give more informative errors than we can
+		// about validation.  Special case strings for nullable
+		// properties ('null') and for resetting to default ('')
+		// - For arrays, make an effort: otherwise the REST API
+		// may interpret a scalar string as a list of length 1
+		// (via one_or_many_property).
+
+		if meta.Nullable && value == "null" {
+			// Nullable types may be explicitly set to null
+			upsert[key] = nil
+		} else if meta.Type != "string" && (value == "") {
+			// Non-string types that receive an empty string
+			// are reset to default
+			remove = append(remove, key)
+		} else if meta.Type == "array" {
+			var a anySlice
+			err := yaml.Unmarshal([]byte(value), &a)
+			if err != nil {
+				return nil, nil, fmt.Errorf("invalid list syntax: %v", err)
+			}
+			upsert[key] = a
+		} else {
+			upsert[key] = value
+		}
+	}
+
+	return upsert, remove, nil
+}
+
+func setCloudConfig(cmd *cobra.Command, cfg *config.Config, p *config.RpkProfile, configs []string) (*controlplanev1.UpdateClusterOperation, error) {
+	cloudClient := publicapi.NewCloudClientSet(cfg.DevOverrides().PublicAPIURL, p.CurrentAuth().AuthToken)
+
+	redpandaConfigs := make(map[string]interface{})
+	paths := make([]string, 0)
+
+	for _, config := range configs {
+		split := strings.SplitN(config, "=", 2)
+		key, value := split[0], split[1]
+		redpandaConfigs[key] = value
+		paths = append(paths, fmt.Sprintf("cluster_configuration.custom_properties.%s", key))
+	}
+	customerProperties, err := structpb.NewStruct(redpandaConfigs)
+	if err != nil {
+		return nil, fmt.Errorf("internal error while converting config to redpanda cloud configs: %v", err)
+	}
+	req := &controlplanev1.UpdateClusterRequest{
+		Cluster: &controlplanev1.ClusterUpdate{
+			Id: p.CloudCluster.ClusterID,
+			ClusterConfiguration: &controlplanev1.ClusterUpdate_ClusterConfiguration{
+				CustomProperties: customerProperties,
+			},
 		},
+		UpdateMask: &fieldmaskpb.FieldMask{Paths: paths},
 	}
 
-	return cmd
+	operation, err := cloudClient.Cluster.UpdateCluster(cmd.Context(), connect.NewRequest(req))
+	if err != nil {
+		var ce *connect.Error
+		if errors.As(err, &ce) {
+			if ce.Code() == connect.CodePermissionDenied {
+				return nil, fmt.Errorf("permission denied")
```

Comment:
How does it look if you don't do the error type assertion? Does the error contain anything more than `permission denied`?

My concern is that the error doesn't indicate what's left to do for the user

Reply:
I updated the text, let me know what do you think